submissions/vakulinina/js-dom/script.js

```diff
@@ -0,0 +1,103 @@
+    ratio: '1 shot of espresso',
+    cup: '2-4 oz. Espresso Cup'
+  },
+  {
+    name: 'flat-white',
+    imageUrl: 'img/flat-white.png',
+    description: 'A flat white also originates from New Zealand and Australia and is very similar to a cappuccino but lacks the foam layer and chocolate powder. To keep the drink creamy rather than frothy, steamed milk from the bottom of the jug is used instead of from the top.',
+    ratio: '1 shot of espresso + 4 oz. of steamed milk',
+    cup: '6 oz. Glass Tumbler'
+  },
+  {
+    name: 'macchiato',
+    imageUrl: 'img/macchiato.png',
+    description: 'The word "macchiato" means mark or stain. This is in reference to the mark that steamed milk leaves on the surface of the espresso as it is dashed into the drink. Flavoring syrups are often added to the drink according to customer preference.',
+    ratio: '1 shot of espresso + 1 to 2 teaspoons of steamed milk',
+    cup: '3 oz. Glass Espresso Cup'
+  },
+  {
+    name: 'mocha',
+    imageUrl: 'img/mocha.png',
+    description: 'The mocha is considered a coffee and hot chocolate hybrid. The chocolate powder or syrup gives it a rich and creamy flavor and cuts the acidity of the espresso.',
+    ratio: '1 shot of espresso + 1-2 oz. of chocolate syrup/powder + 1-3 oz. of steamed milk + 2-3 cm of foam or whipped cream',
+    cup: '6-8 oz. Irish Coffee Mug'
+  },
+  {
+    name: 'red-eye',
+    imageUrl: 'img/red-eye.png',
+    description: 'The red eye\'s purpose is to add a boost of caffeine to your standard cup of coffee.',
+    ratio: '1 shot of espresso + 6 oz. of drip-brewed coffee',
+    cup: '8 oz. Coffee Mug'
+  },
+  {
+    name: 'vienna',
+    imageUrl: 'img/vienna.png',
+    description: 'There are a few variations on the Vienna, but one of the most common is made with two ingredients: espresso and whipped cream. The whipped cream takes the place of milk and sugar to provide a creamy texture.',
+    ratio: '1-2 shots of espresso + 2 oz. of whipped cream',
+    cup: '4-5 oz. Espresso Mug'
+  }
+];
+
+const navButtons = document.querySelector('.nav-buttons');
+const itemDescription = document.querySelector('.description');
+const itemImage = document.querySelector('.item-image');
+const itemRecipe = document.querySelector('.recipe');
+const itemTitle = document.querySelector('h2');
+const menuButton = document.querySelector('.menu-button');
+const navBar = document.querySelector('.nav-bar');
+
+const hideMenu = () => {
+  if (navBar.classList.contains('visible')) {
+    navBar.classList.remove('visible')
+  }
+}
+
+const getProp = (id, prop) => coffeeDrinks.filter(obj => obj.name === id)[0][prop];
+
+menuButton.addEventListener('click', function () {
+  navBar.classList.toggle('visible');
+})
+
+navButtons.addEventListener('click', function (evt) {
+  if (evt.target.tagName === 'BUTTON') {
+    const id = evt.target.id;
+    hideMenu();
+    itemTitle.textContent = getProp(id, 'name').toUpperCase();
```

Comment:
Look how many times you called the `getProp` in this function. It seems 5 times. 
So 5 times you filter the `coffeeDrinks` list and take the necessary property from the element.
I'd recommend using `Array.prototype.find` for finding necessary object by the `id` from the `coffeeDrinks` list and then just get necessary properties from it.
It can help you to increase the performance of your app.